remove extra spaces in regexp, fix version listing

niklas-sparfeld opened the pull request:
This change fixes version listing: `asdf list-all terragrunt`

The output of `curl -s https://api.github.com/repos/gruntwork-io/terragrunt/releases` does not contain spaces after `"tag_name":`, which is why the returned list was always emtpy.

This PR also fixes other asdf features that depend on the version, such as `asdf install terragrunt latest`.

Have a nice day :-)

lotia commented:
Thanks for submitting the PR. The build is failing. Do you have insight into why?

lotia commented:
Can you please rebase this on to master now that I've merged in #4 , happy to review after and apologies for the slow response.

lotia commented:
Also are you testing on Linux or macOS? I'm running `asdf list-all terragrunt` and I do have output when I run on macOS on fish shell. I see the following output with asdf plugin version
```
terragrunt  https://github.com/lotia/asdf-terragrunt.git  master  0cd53a5
```

```
# asdf list all terragrunt
0.19.20
0.19.21
0.19.22
0.19.23
0.19.24
0.19.25
0.19.26
0.19.27
0.19.28
0.19.29
0.19.30
0.19.31
0.20.0
0.20.1
0.20.2
0.20.3
0.20.4
0.20.5
0.21.0
0.21.1
0.21.2
0.21.3
0.21.4
0.21.5
0.21.6
0.21.7
0.21.8
0.21.9
0.21.10
0.21.11
```
or 

```
# asdf list-all terragrunt
0.19.20
0.19.21
0.19.22
0.19.23
0.19.24
0.19.25
0.19.26
0.19.27
0.19.28
0.19.29
0.19.30
0.19.31
0.20.0
0.20.1
0.20.2
0.20.3
0.20.4
0.20.5
0.21.0
0.21.1
0.21.2
0.21.3
0.21.4
0.21.5
0.21.6
0.21.7
0.21.8
0.21.9
0.21.10
0.21.11
```

niklas-sparfeld-gcx committed:
improve resilience of version parser

niklas-sparfeld commented:
Interesting... I am testing on macOS 10.14.6 and I found the difference:
From some old dotfiles, that I migrated from machine to machine through the ages, I had the following in my `.curlrc`:

```
# Disguise as IE 9 on Windows 7.
user-agent = "Mozilla/5.0 (compatible; MSIE 9.0; Windows NT 6.1; Trident/5.0)"
```

This changes the default response of https://api.github.com/repos/gruntwork-io/terragrunt/releases to be minified instead of human readable ;)
Which means that there is no whitespace anymore :)

I changed my PR to work with both cases by making the space optional. I do not know whether you really want to merge this PR... Locally I already removed the line from my `.curlrc`.

lotia commented:
I really appreciate the previous PR where you've made the pipeline significantly more elegant. I'm leaning towards not merging it since it is a workaround/compatability change to work with local curl config rather than the default. I suspect this may lead us down the path of many workarounds. I hope you understand my leaning towards not merging this one. Thanks agan for #4 and please do repsond if you disagree with me.
